pkg/controller/operators/olm/operator.go

```diff
@@ -1435,7 +1435,9 @@ func (a *Operator) ensureDeploymentAnnotations(logger *logrus.Entry, csv *v1alph
 	// compare deployments to see if any need to be created/updated
 	existingMap := map[string]*appsv1.Deployment{}
 	for _, d := range existingDeployments {
```

Comment:
Instead of creating two maps, can we use the `depName` map while iterating through the existing deployment list and check if the existing deployment name is in the map?

```go
for _, dep := range existingDeployments {
         if _, ok := depNames[dep.GetName()]; !ok {
               continue
         }
          ...
}
```